src/Confluent.Kafka/ConsumeResult.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2016-2017 Confluent Inc., 2015-2016 Andreas Heider
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Derived from: rdkafka-dotnet, licensed under the 2-clause BSD License.
+//
+// Refer to LICENSE for more information.
+
+
+namespace Confluent.Kafka
+{
+    /// <summary>
+    ///     Represents a message consumed from kafka cluster.
+    /// </summary>
+    public class ConsumeResult<TKey, TValue> : MessageAndMetadata<TKey, TValue>
+    {
+        /// <summary>
+        ///     True if the ConsumeResult represents an end-of-partition event,
+        ///     false otherwise.
+        /// </summary>
+        public bool IsPartitionEOF { get => Error.Code == ErrorCode.Local_PartitionEOF; }
```

Comment:
Why not expression bodied syntax?

`public bool IsPartitionEOF => Error.Code == ErrorCode.Local_PartitionEOF;` looks more expressive.

Reply:
yep, though I've changed this now such that `ConsumeResult` doesn't contain an error field and this isn't derived.